src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -747,8 +752,20 @@ public Builder setMBeanName(final String mBeanName) {
     public void execute(Runnable runnable) {
         Assert.checkNotNullParam("runnable", runnable);
         final Runnable realRunnable = JBossExecutors.classLoaderPreservingTaskUnchecked(runnable);
-        int result;
-        result = tryExecute(realRunnable);
+        final int result;
+        try (TryExecuteTask tryExecuteTask = POOLED_COMBINER_TASKS.get().on(this, realRunnable)) {
+            tryExecuteCombiner.combine(tryExecuteTask, idleCount -> {
+                final int spinWait = PARK_SPINS - YIELD_FACTOR;
+                if (idleCount >= spinWait) {
+                    Thread.yield();
+                } else {
+                    JDKSpecific.onSpinWait();
+                    idleCount++;
+                }
+                return idleCount;
+            });
```

Comment:
I think we can extract this IdleStrategy to a singleton and avoid allocation cost.